### PR TITLE
doc: improve top-level README links

### PR DESCRIPTION
**Pull request by coryan**
Link directly to the quickstart guides from the top-level README. Also
include brief "make install" instructions, with some caveats, and links
to the more detailed instructions.

---
This change is [Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/5216)

### CODE REVIEW

**Review by devjgm — APPROVED** [review]

File: ci/generate-markdown/generate-readme.sh [hunk 1]
```diff
@@ -67,8 +67,23 @@ See each library's `README.md` file for more information about:
...
+```sh
+cmake -H. -Bcmake-out
+cmake --build cmake-out
+sudo cmake --build cmake-out --target install
+```
+
+You can find detailed instructions on how to install and/or compile all the
+dependencies for several plaforms in the [packaging guid](doc/packaging.md).
+
+For application developers that prefer to build from source, the quickstart
```
Comment (devbww) [inline ci/generate-markdown/generate-readme.sh:84]: s/that/who/

File: ci/generate-markdown/generate-readme.sh [hunk 2]
```diff
@@ -67,8 +67,23 @@ See each library's `README.md` file for more information about:
...
+guides for each library (see above) include instructions on how to incorporate
+the library into your CMake-based or Bazel-based builds.
```
Comment (devbww) [inline ci/generate-markdown/generate-readme.sh:86]: s/your/their/

**Review by devbww — APPROVED** [review]

**Comment by mr-salty** [comment]
___
*[README.md, line 57 at r1](https://reviewable.io/reviews/googleapis/google-cloud-cpp/5216#-MJCvVQE66z4BId7cyiM:-MJCvVQE66z4BId7cyiN:b-trq8w6) ([raw file](https://github.com/googleapis/google-cloud-cpp/blob/6f44e38c34fcf60508e1c9fa8ad420178bd12811/README.md#L57)):*
> ```Markdown
> 
> You can find detailed instructions on how to install and/or compile all the
> dependencies for several plaforms in the [packaging guid](doc/packaging.md).
> ```

s/guid/guide

is it worth trying to get spellcheck to work on these files?

**Review by mr-salty — APPROVED** [review]
:lgtm: thanks

> Reviewed 2 of 2 files at r1.
> *[Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/5216#-:-MJCwIaX1FcJVmTw-I5N:b-v6ynew)* status: all files reviewed, 6 unresolved discussions (waiting on @coryan)

___
*[README.md, line 59 at r1](https://reviewable.io/reviews/googleapis/google-cloud-cpp/5216#-MJCw49-Cr-ganFfLFV8:-MJCw4904qRu9ZyFy45_:b-qfgyph) ([raw file](https://github.com/googleapis/google-cloud-cpp/blob/6f44e38c34fcf60508e1c9fa8ad420178bd12811/README.md#L59)):*
> ```Markdown
> dependencies for several plaforms in the [packaging guid](doc/packaging.md).
> 
> For application developers that prefer to build from source, the quickstart
> ```

dup'ing bww's comment from the other file: s/that/who
___
*[README.md, line 74 at r1](https://reviewable.io/reviews/googleapis/google-cloud-cpp/5216#-MJCvvrEAxQKt94kJGPA:-MJCvvrEAxQKt94kJGPB:beas4ri) ([raw file](https://github.com/googleapis/google-cloud-cpp/blob/6f44e38c34fcf60508e1c9fa8ad420178bd12811/README.md#L74)):*
> ```Markdown
> Each library (linked above) contains a directory named `quickstart/` that's
> intended to help you get up and running in a matter of minutes. This
> `quickstart/` directory contains a minimal "Hello World" program demonstrating
> ```

mention the links here?
___
*[ci/generate-markdown/generate-readme.sh, line 82 at r1](https://reviewable.io/reviews/googleapis/google-cloud-cpp/5216#-MJCwBaS99jtTwpBh48j:-MJCwBaT9usIVVKPZZ4-:bu029ef) ([raw file](https://github.com/googleapis/google-cloud-cpp/blob/6f44e38c34fcf60508e1c9fa8ad420178bd12811/ci/generate-markdown/generate-readme.sh#L82)):*
> ```Shell
> 
> You can find detailed instructions on how to install and/or compile all the
> dependencies for several plaforms in the [packaging guid](doc/packaging.md).
> ```

s/guid/guide

**Review by coryan — COMMENTED** [review]
Rebased and addressed comments

> *[Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/5216#-:-MJDYyDZ82yavXGNKHcr:bbinqij)* status: 0 of 2 files reviewed, 6 unresolved discussions (waiting on @devbww and @mr-salty)

___
*[README.md, line 57 at r1](https://reviewable.io/reviews/googleapis/google-cloud-cpp/5216#-MJCvVQE66z4BId7cyiM:-MJDWocx9RFUCksesV2a:b-kskv4u) ([raw file](https://github.com/googleapis/google-cloud-cpp/blob/6f44e38c34fcf60508e1c9fa8ad420178bd12811/README.md#L57)):*
<details><summary><i>Previously, mr-salty (Todd Derr) wrote…</i></summary><blockquote>

s/guid/guide

is it worth trying to get spellcheck to work on these files?
</blockquote></details>

Done.  FWIW, the spell checker in my IDE is happy (modulo `Kokoro` and a few other code-things).
___
*[README.md, line 59 at r1](https://reviewable.io/reviews/googleapis/google-cloud-cpp/5216#-MJCw49-Cr-ganFfLFV8:-MJDWoxp1SIVgB_SN03j:b-896fix) ([raw file](https://github.com/googleapis/google-cloud-cpp/blob/6f44e38c34fcf60508e1c9fa8ad420178bd12811/README.md#L59)):*
<details><summary><i>Previously, mr-salty (Todd Derr) wrote…</i></summary><blockquote>

dup'ing bww's comment from the other file: s/that/who
</blockquote></details>

Done.
___
*[README.md, line 74 at r1](https://reviewable.io/reviews/googleapis/google-cloud-cpp/5216#-MJCvvrEAxQKt94kJGPA:-MJDYuo1B6ZTjoNSBmu-:bpzh2v0) ([raw file](https://github.com/googleapis/google-cloud-cpp/blob/6f44e38c34fcf60508e1c9fa8ad420178bd12811/README.md#L74)):*
<details><summary><i>Previously, mr-salty (Todd Derr) wrote…</i></summary><blockquote>

mention the links here?
</blockquote></details>

Is this what you meant?
___
*[ci/generate-markdown/generate-readme.sh, line 82 at r1](https://reviewable.io/reviews/googleapis/google-cloud-cpp/5216#-MJCwBaS99jtTwpBh48j:-MJDWqAMDJNtu1nxo04F:b-896fix) ([raw file](https://github.com/googleapis/google-cloud-cpp/blob/6f44e38c34fcf60508e1c9fa8ad420178bd12811/ci/generate-markdown/generate-readme.sh#L82)):*
<details><summary><i>Previously, mr-salty (Todd Derr) wrote…</i></summary><blockquote>

s/guid/guide
</blockquote></details>

Done.
___
*[ci/generate-markdown/generate-readme.sh, line 84 at r1](https://reviewable.io/reviews/googleapis/google-cloud-cpp/5216#-MJCcMNAMW15CGAD7m0r-r1-84:-MJDWqWgDl2_qAQImXZC:b-896fix) ([raw file](https://github.com/googleapis/google-cloud-cpp/blob/6f44e38c34fcf60508e1c9fa8ad420178bd12811/ci/generate-markdown/generate-readme.sh#L84)):*
<details><summary><i>Previously, devbww (Bradley White) wrote…</i></summary><blockquote>

s/that/who/
</blockquote></details>

Done.
___
*[ci/generate-markdown/generate-readme.sh, line 86 at r1](https://reviewable.io/reviews/googleapis/google-cloud-cpp/5216#-MJCcMNAMW15CGAD7m0r-r1-86:-MJDWrKNE2TyaHmRa1OA:b-896fix) ([raw file](https://github.com/googleapis/google-cloud-cpp/blob/6f44e38c34fcf60508e1c9fa8ad420178bd12811/ci/generate-markdown/generate-readme.sh#L86)):*
<details><summary><i>Previously, devbww (Bradley White) wrote…</i></summary><blockquote>

s/your/their/
</blockquote></details>

Done.

**Review by mr-salty — CHANGES_REQUESTED** [review]
:lgtm:

> Reviewed 2 of 2 files at r2.
> *[Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/5216#-:-MJDkh447aJEnmbh-O3O:bnfp4nl)* status: all files reviewed, 6 unresolved discussions (waiting on @coryan, @devbww, and @mr-salty)

___
*[README.md, line 74 at r1](https://reviewable.io/reviews/googleapis/google-cloud-cpp/5216#-MJCvvrEAxQKt94kJGPA:-MJDkSLC2w3fZamETNJ9:b-dbamwc) ([raw file](https://github.com/googleapis/google-cloud-cpp/blob/6f44e38c34fcf60508e1c9fa8ad420178bd12811/README.md#L74)):*
<details><summary><i>Previously, coryan (Carlos O'Ryan) wrote…</i></summary><blockquote>

Is this what you meant?
</blockquote></details>

I was thinking referring to the top of the doc but I think just duping the links here like you did is better.

**Review by mr-salty — APPROVED** [review]
:lgtm:

> *[Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/5216#-:-MJDkjDl74QWNfSdPU9M:bnfp4nl)* status: all files reviewed, 6 unresolved discussions (waiting on @coryan, @devbww, and @mr-salty)

**Pull request merged**